forc-plugins/forc-debug/src/server/handlers/handle_launch.rs

```diff
@@ -0,0 +1,216 @@
+use crate::server::AdapterError;
+use crate::server::DapServer;
+use forc_pkg::{self, BuildProfile, Built, BuiltPackage, PackageManifestFile};
+use forc_test::execute::{DebugResult, TestExecutor};
+use forc_test::setup::TestSetup;
+use forc_test::BuiltTests;
+use std::{cmp::min, collections::HashMap, fs, path::PathBuf, sync::Arc};
+use sway_types::span::Position;
+
+impl DapServer {
+    /// Handles a `launch` request. Returns true if the server should continue running.
+    pub fn handle_launch(&mut self) -> Result<bool, AdapterError> {
+        // Build tests for the given path.
+        let (pkg_to_debug, test_setup) = self.build_tests()?;
+        let entries = pkg_to_debug.bytecode.entries.iter().filter_map(|entry| {
+            if let Some(test_entry) = entry.kind.test() {
+                return Some((entry, test_entry));
+            }
+            None
+        });
+
+        // Construct a TestExecutor for each test and store it
+        let executors: Vec<TestExecutor> = entries
+            .filter_map(|(entry, test_entry)| {
+                let offset = u32::try_from(entry.finalized.imm)
+                    .expect("test instruction offset out of range");
+                let name = entry.finalized.fn_name.clone();
+                if test_entry.file_path.as_path() != self.state.program_path.as_path() {
+                    return None;
+                }
+
+                Some(TestExecutor::new(
+                    &pkg_to_debug.bytecode.bytes,
+                    offset,
+                    test_setup.clone(),
+                    test_entry,
+                    name.clone(),
+                ))
+            })
+            .collect();
+        self.state.init_executors(executors);
+
+        // Start debugging
+        self.state.update_vm_breakpoints();
+        while let Some(executor) = self.state.executors.first_mut() {
```

Comment:
This looks similar to the logic above in the `handle_continue` function. Can it be abstracted away and called from both places?